CIoTS/generator.py

```diff
@@ -64,20 +70,30 @@ def generate_stable_graph(self):
 
     def _generate_graph(self):
         self.graph = nx.DiGraph()
-        node_ids = np.array([[(d, l) for l in range(self.max_p, -1, -1)] for d in range(self.dimensions)])
-        self.graph.add_nodes_from([node_name(d, l) for d, l in
-                                   np.reshape(node_ids, (self.dimensions * self.length, 2))])
+        node_ids = list(product([ d for d in range(self.dimensions)],
+                                [l for l in range(self.max_p, -1, -1)]))
+        self.graph.add_nodes_from([node_name(*n) for n in node_ids])
 
         # Ensure max_p is utilized
         d_t = np.random.choice(self.dimensions)
         d_p = np.random.choice(self.dimensions)
-        self.graph.add_edge(node_name(d_p, self.max_p), node_name(d_t, 0))
 
         for d in range(self.dimensions):
             # Generate random edges to previous nodes
-            candidates = np.delete(np.reshape(node_ids[:, :-1], (-1, 2)), [d_p * self.max_p], axis=0)
-            remaining_edges = self.incoming_edges - (1 if d_t == d else 0)
-            picks = candidates[np.random.choice(candidates.shape[0], remaining_edges, replace=False)]
+            candidates = [n for n in node_ids if n[1] > 0]
+            remaining_edges = self.incoming_edges
+
+            if d == d_t:
+                remaining_edges -= 1
+                candidates.remove((d_p, self.max_p))
+                self.graph.add_edge(node_name(d_p, self.max_p), node_name(d, 0))
+
+            # autocorrelation  
+            if self.enforce_autocorrelation:
+                candidates.remove((d, 1))
+                self.graph.add_edge(node_name(d, 1), node_name(d, 0))
```

Comment:
What's going to be the weight of the autocorrelation? I initially assumed it to be 1 as default, but that seems to be not the case. In either case, I think it would be better to have the param `enfore_autocorrelation` as a float with default value 0, then we can control that.